posixlib/src/main/resources/scala-native/time.c

```diff
@@ -91,8 +112,24 @@ size_t scalanative_strftime(char *buf, size_t maxsize, const char *format,
     return strftime(buf, maxsize, format, &tm);
 }
 
+// XSI
+char *scalanative_strptime(const char *s, const char *format,
+                           struct scalanative_tm *scala_tm) {
+    // strptime is known to not set is_dst field for %Z format.
+    // Take runtime hit of clearing entire structure to be robust
+    // to that and undiscovered corner cases where strptime does not fill
+    // a field under some condition.
+    struct tm tm = {0};
+
+    char *result = strptime(s, format, &tm);
+    scalanative_tm_init(scala_tm, &tm);
```

Comment:
These seem out of order.

Reply:
Eric,  thank you for the review.

I believe the code is correct, but please double check my logic. 

C strptime() takes a Cstring src uses a Cstring format to convert 
the string into a C tm.  Then the next line copies the C tm 
to a scalanative_tm, which gets passed back to the Scala caller.

I can add a short comment on the sn_tm_init() line if you can suggest one.
Or I can correct a grievous error if he same person who wrote the code
is blind to his own mistake.  I certainly do not want to be publishing flaws.